jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
But doesn't L265 then allow very strange behaviour? For example, consider a case of two datasets (jyu-lb engine does not allow this, but this particular method does not know it, or it is programmed as there could be several datasets), and then the user/client changes, at some point, the dataset names equal (outside the engine). Then, after execution of this method, only the other will be stored in SD?

In general, this whole approach of storing the datasets in a dictionary with their names used as the key (if they are supposed to be the "real", "dynamic" names of the datasets), seems very error prone ...